apiserver/facades/controller/crosscontroller/crosscontroller.go

```diff
@@ -32,7 +33,18 @@ func NewStateCrossControllerAPI(ctx facade.Context) (*CrossControllerAPI, error)
 	st := ctx.State()
 	return NewCrossControllerAPI(
 		ctx.Resources(),
-		func() ([]string, string, error) { return common.StateControllerInfo(st) },
+		func() ([]string, string, error) {
+			addrs, caCert, err := common.StateControllerInfo(st)
```

Comment:
Does the public DNS address route to a ha proxy in front of the controller? Is this the correct CA Cert to use with the DNS name?

Reply:
It would be the wrong cert, but the right CA cert should come from the system certificates as it will be signed by a "proper" CA.

Reply:
Is there an additional change needed? Or is it the case we pass across the cert that works with the "internal" IP addresses and the system certs will get used automatically with the dns name?